common/plugins/eu.esdihumboldt.hale.common.schema.persist/src/eu/esdihumboldt/hale/common/schema/persist/hsd/json/JsonToSchema.groovy

```diff
@@ -197,11 +206,11 @@ public class JsonToSchema {
 						Object constraint = desc.getFactory().restore(config, definition, typeProvider, classResolver)
 						definition.setConstraint(constraint)
 					} catch (Exception e) {
-						reporter.error(new IOMessageImpl("Failed to restore constraint of type $id", e))
+						error("Failed to restore constraint of type $id", e)
 					}
 				}
 				else {
-					reporter.error(new IOMessageImpl("Could not find factory for constraint with type $id"))
+					warn("Could not find factory for constraint with type $id")
```

Comment:
Why would an error message prevent loading the HSD? Wouldn't that just show up as error in the report?

Reply:
Sorry, I'll try to clarify:

The problem in #707 (HSD not loading) was caused by using a non-existent constructor of `IOMessageImpl`. This was fixed by using the local `error()` method instead.

In a second step I changed the message level to warning to allow transformations to succeed even if the constraints could not be recreated. In the case of FGDB constraints this would prevent successful transformation of projects that use HSDs created from an FGDB.